File: 2024/01/java/src/main/java/net/advw/aoc/Part1.java
package net.advw.aoc;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Part1 {
    public static String solve(String input) {
        // Parse input into two lists
        String[] lines = input.trim().split("\n");
        List<Integer> leftList = new ArrayList<>();
        List<Integer> rightList = new ArrayList<>();

        for (String line : lines) {
            String[] parts = line.trim().split("\\s+");
            if (parts.length == 2) {
                leftList.add(Integer.parseInt(parts[0]));
                rightList.add(Integer.parseInt(parts[1]));
            }
        }

        // Sort both lists
        Collections.sort(leftList);
        Collections.sort(rightList);

        // Calculate total distance
        int totalDistance = 0;
        for (int i = 0; i < leftList.size(); i++) {
            int distance = Math.abs(leftList.get(i) - rightList.get(i));
            totalDistance += distance;
        }

        return String.valueOf(totalDistance);
    }
}

File: 2024/01/go/part1.go
package main

import (
	"fmt"
	"math"
	"sort"
	"strconv"
	"strings"
)

func solvePart1(in string) string {
	// Parse input into two slices
	var left, right []int
	lines := strings.Split(strings.TrimSpace(in), "\n")

	for _, line := range lines {
		var l, r int
		// Using sscanf to parse two numbers separated by whitespace
		n, _ := fmt.Sscanf(line, "%d %d", &l, &r)
		if n == 2 {
			left = append(left, l)
			right = append(right, r)
		}
	}

	// Sort both slices
	sort.Ints(left)
	sort.Ints(right)

	// Calculate total distance
	totalDistance := 0
	for i := 0; i < len(left); i++ {
		distance := int(math.Abs(float64(left[i] - right[i])))
		totalDistance += distance
	}

	return strconv.Itoa(totalDistance)
}
